src/sync.cpp

```diff
@@ -468,18 +496,37 @@ namespace librealsense
 
     void timestamp_composite_matcher::update_last_arrived(frame_holder& f, matcher* m)
     {
+        if(f->supports_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS))
+            _fps[m] = f->get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS);
+
+        else
+            _fps[m] = f->get_stream()->get_framerate();
+
         _last_arrived[m] = std::chrono::duration<double, std::milli>(std::chrono::system_clock::now().time_since_epoch()).count();
     }
 
+    int timestamp_composite_matcher::get_fps(const frame_holder & f)
+    {
+        auto fps = 0;
+        if(f.frame->supports_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS))
+        {
+            fps = f.frame->get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS);
+        }
+        LOG_DEBUG("fps " <<fps<<" "<< frame_to_string(const_cast<frame_holder&>(f)));
+        return fps?fps:f.frame->get_stream()->get_framerate();
+    }
+
     void timestamp_composite_matcher::update_next_expected(const frame_holder & f)
     {
-        auto fps = f.frame->get_stream()->get_framerate();
+        auto fps = get_fps(f);
         auto gap = 1000 / fps;
```

Comment:
gap is calculated inconsistently - in this occurrence it is trimmed to int. 
In another place  -auto gap = 1000.f/ (float)get_fps(*synced_frame);